src/librustdoc/core.rs

```diff
@@ -140,27 +137,26 @@ impl<'tcx> DocContext<'tcx> {
     /// [`RefCell`]: std::cell::RefCell
     /// [`Debug`]: std::fmt::Debug
     /// [`clean::Item`]: crate::clean::types::Item
-    crate fn next_def_id(&self, crate_num: CrateNum) -> DefId {
-        let mut fake_ids = self.fake_def_ids.borrow_mut();
-
-        let def_index = match fake_ids.entry(crate_num) {
-            Entry::Vacant(e) => {
-                let num_def_idx = {
-                    let num_def_idx = if crate_num == LOCAL_CRATE {
-                        self.tcx.hir().definitions().def_path_table().num_def_ids()
-                    } else {
-                        self.enter_resolver(|r| r.cstore().num_def_ids(crate_num))
-                    };
-
-                    DefIndex::from_usize(num_def_idx)
+    crate fn next_def_id(&mut self, crate_num: CrateNum) -> DefId {
+        let def_index = if let Some(def_idx) = self.fake_def_ids.get_mut(&crate_num) {
```

Comment:
You forgot to actually remove the `RefCell`, so this resolves to `RefCell::get_mut` instead of `HashMap::get_mut`.

You said the entry API gave you borrowck errors - did you forget to commit some code?

Reply:
Yeah, looks like somehow I didn't commit the removal of `RefCell`. It's weird because I only used `git add src/librustdoc/{...}` and the changes compilde locally.